hoomd/compute.py

```diff
@@ -191,71 +86,148 @@ class thermo(_compute):
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def temperature(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getTemperature()
+        else:
+            return None
 
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_compute = _hoomd.ComputeThermo(hoomd.context.current.system_definition, group.cpp_group, suffix);
+    @log
+    def pressure(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressure()
         else:
-            self.cpp_compute = _hoomd.ComputeThermoGPU(hoomd.context.current.system_definition, group.cpp_group, suffix);
+            return None
 
-        hoomd.context.current.system.addCompute(self.cpp_compute, self.compute_name);
+    @log
+    def pressureXX(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureXX()
+        else:
+            return None
 
-        # save the group for later referencing
-        self.group = group;
-        # add ourselves to the list of compute thermos specified so far
-        hoomd.context.current.thermos.append(self);
+    @log
+    def pressureXY(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureXY()
+        else:
+            return None
 
-    def disable(self):
-        R""" Disables the thermo.
+    @log
+    def pressureXZ(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureXZ()
+        else:
+            return None
 
-        Examples::
+    @log
+    def pressureYY(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureYY()
+        else:
+            return None
 
-            my_thermo.disable()
+    @log
+    def pressureYZ(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureYZ()
+        else:
+            return None
 
-        Executing the disable command will remove the thermo compute from the system. Any ```hoomd.run``` command
-        executed after disabling a thermo compute will not be able to log computed values with ``hoomd.analyze.log``.
+    @log
+    def pressureZZ(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureZZ()
+        else:
+            return None
 
-        A disabled thermo compute can be re-enabled with :py:meth:`enable()`.
-        """
+    @log
+    def kineticEnergy(self):
```

Comment:
```suggestion
    def kinetic_energy(self):
```
And similarly use PEP8 names throughout.

Reply:
:+1: